pkg/kv/kvserver/rangefeed/unbuffered_registration.go

```diff
@@ -363,8 +365,41 @@ func (ubr *unbufferedRegistration) maybeRunCatchUpScan(ctx context.Context) erro
 		ubr.metrics.RangeFeedCatchUpScanNanos.Inc(timeutil.Since(start).Nanoseconds())
 	}()
 
-	return catchUpIter.CatchUpScan(ctx, ubr.stream.SendUnbuffered, ubr.withDiff, ubr.withFiltering,
-		ubr.withOmitRemote)
+	var out func(event *kvpb.RangeFeedEvent) error
+	var after func() error
+
+	if ubr.bulkDelivery {
+		var size int
+		var buf []*kvpb.RangeFeedEvent
+		out = func(event *kvpb.RangeFeedEvent) error {
+			buf = append(buf, event)
+			size += event.Size()
+			if len(buf) >= 1<<12 || size > 1<<20 {
+				if err := ubr.stream.SendUnbuffered(&kvpb.RangeFeedEvent{BulkEvents: &kvpb.RangeFeedBulkEvents{Events: buf}}); err != nil {
```

Comment:
Anyway, left a few minor comments.  We've gone back and forth on how much to care about duplication between buffered and unbuffered sender. In this case I think I'm OK with the duplication, but perhaps it is worth a reminder comment in both call sites to update the other one if we need a change here.  I suppose we could pull out the size limits to some const/vars as well.